lib/config/providers/group_provider.dart

```diff
@@ -675,38 +653,85 @@ class GroupsNotifier extends Notifier<GroupsState> {
     }
   }
 
-  /// Load members for all groups (used during initial group loading)
-  Future<void> _loadMembersForAllGroups(List<Group> groups) async {
+  /// Load group metadata lazily, processing each group one at a time
```

Comment:
@codeswot instead of loading one group at a time (sequentially) this could be made faster by Future.wait()-ing multiple groups at a time. Probably 5 - 7 groups. There should be a significant improvement in the speed. WDYT?